src/back/wgsl/writer.rs

```diff
@@ -1604,8 +1604,7 @@ impl<W: Write> Writer<W> {
                             TypeInner::Scalar {
                                 kind: crate::ScalarKind::Bool,
                                 ..
-                            }
-                            | TypeInner::Vector { .. } => "!",
+                            } => "!",
```

Comment:
With this change boolean vectors are now using `~` instead of `!`. All backends use the code below (that we could also use here):
https://github.com/gfx-rs/naga/blob/3c7dbc4016b84a35c69b30305b12abaeefc21fd9/src/back/msl/writer.rs#L1539-L1543

Ideally we should just add another variant to `UnaryOperator` for `LogicalNot` and rename `Not` to `BitwiseNot`.